frameworks/C/octane/src/program.cpp

```diff
@@ -0,0 +1,49 @@
+#include <stdio.h>
+#include <uv.h>
+#include <stdbool.h>
+#include <octane.h>
+#include "responders/sds_responder.hpp"
+
+void timer_callback(uv_timer_t* timer);
+
+void on_request(http_connection* connection, http_request** requests, int number_of_requests);
+
+char* current_time;
+uv_timer_t timer;
+
+int main(int argc, char *argv[]) {
+    http_listener* listener = new_http_listener();
+    uv_timer_init(listener->loop, &timer);
+    uv_timer_start(&timer, timer_callback, 0, 500);
+
+    begin_listening(listener, "0.0.0.0", 8000, false, 4, 128, NULL, NULL, NULL, on_request);
+
+    printf("Listening...\n");
+}
+
+void on_request(http_connection* connection, http_request** requests, int number_of_requests) {
+    write_batch* batch = create_write_batch(number_of_requests);
+
+    for (int i=0; i<number_of_requests; i++) {
+        if (requests[i]->path[1] == 'p') {
+            create_plaintext_response_sds(batch);
+        } else if (requests[i]->path[1] == 'j') {
```

Comment:
I think this is really nit picking because not all uses of HTTP libraries require routing and some HTTP libraries shouldn't dictate how routing syntax behaves either.

Also the benchmark doesn't even stress routing, for example I could implement a common radix trie just to satisfy this request but the code is essentially at runtime going to do the _exact_ same thing as this code. Let's be honest, everyone's routing library in this benchmark is comparing one character or multi-character single instruction comparisons and then routing.

So I'll implement this only to satisfy something that takes several hundreds of lines of code that essentially executes a single character comparison because there's no routes that share the same prefix.

This code does not give me any special performance over anyone else because everyones is doing the same thing in the end.